src/Deedle/FrameUtils.fs

```diff
@@ -67,16 +68,13 @@ module internal Reflection =
 
   /// Helper function used when building frames from data tables
   let createTypedVector : _ -> seq<OptionalValue<obj>> -> _ =
-    let cache = Dictionary<_, _>()
-    (fun typ ->
-      match cache.TryGetValue(typ) with
-      | true, res -> res
-      | false, _ ->
-          let par = Expression.Parameter(typeof<seq<OptionalValue<obj>>>)
-          let body = Expression.Call(createTypedVectorMi.MakeGenericMethod([| typ |]), par)
-          let f = Expression.Lambda<Func<seq<OptionalValue<obj>>, IVector>>(body, par).Compile()
-          cache.Add(typ, f.Invoke)
-          f.Invoke )
+    let cache = ConcurrentDictionary<_, seq<OptionalValue<obj>> -> _>()
+    let valueFactory typ =
```

Comment:
Change this to: 
```
   let valueFactory = 
       Func<_,_> (fun typ -> 
         let par = Expression.Parameter(typeof<seq<OptionalValue<obj>>>)
         let body = Expression.Call(createTypedVectorMi.MakeGenericMethod([| typ |]), par)
         Expression.Lambda<Func<seq<OptionalValue<obj>>, _>>(body, par).Compile().Invoke)
   fun typ ->
     cache.GetOrAdd(typ, valueFactory)    
```

So the `Func<_,_>` call to create the delegate is only executed once